store/sqlstore/post_store.go

```diff
@@ -468,52 +468,48 @@ func (s *SqlPostStore) GetPosts(channelId string, offset int, limit int, allowFr
 	// Caching only occurs on limits of 30 and 60, the common limits requested by MM clients
 	if allowFromCache && offset == 0 && (limit == 60 || limit == 30) {
 		if cacheItem, ok := s.lastPostsCache.Get(fmt.Sprintf("%s%v", channelId, limit)); ok {
-			if s.metrics != nil {
-				s.metrics.IncrementMemCacheHitCounter("Last Posts Cache")
-			}
-
 			return cacheItem.(*model.PostList), nil
-		} else {
-			if s.metrics != nil {
-				s.metrics.IncrementMemCacheMissCounter("Last Posts Cache")
-			}
-		}
-	} else {
-		if s.metrics != nil {
-			s.metrics.IncrementMemCacheMissCounter("Last Posts Cache")
 		}
 	}
 
+	if s.metrics != nil {
+		s.metrics.IncrementMemCacheHitCounter("Last Posts Cache")
+	}
+
 	rpc := s.getRootPosts(channelId, offset, limit)
 	cpc := s.getParentsPosts(channelId, offset, limit)
 
 	var err *model.AppError
 	list := model.NewPostList()
 
-	if rpr := <-rpc; rpr.Err != nil {
-		err = rpr.Err
-	} else if cpr := <-cpc; cpr.Err != nil {
-		err = cpr.Err
-	} else {
-		posts := rpr.Data.([]*model.Post)
-		parents := cpr.Data.([]*model.Post)
+	rpr := <-rpc
+	cpr := <-cpc
```

Comment:
This is not the same, not a big difference, but you are waiting for both here, and if the first one fail before, you don't need to wait for the second one, my suggestion would be something like:

```go
rpr := <-rpc
if rpr.Err != nil {
    return nil, rpr.Err
}
cpr := <-cpc
if cpr.Err != nil {
    return nil, cpr.Err
}
```